test/api_test.go

```diff
@@ -138,51 +186,54 @@ func TestAPIcensusAndVote(t *testing.T) {
 
 	vote := &models.VoteEnvelope{
 		Nonce:       util.RandomBytes(16),
-		ProcessId:   election.ElectionID,
+		ProcessId:   te.election.ElectionID,
 		VotePackage: votePackageBytes,
 	}
 	vote.Proof = &models.Proof{
 		Payload: &models.Proof_Arbo{
 			Arbo: &models.ProofArbo{
 				Type:       models.ProofArbo_BLAKE2B,
-				Siblings:   censusData.Proof,
-				LeafWeight: censusData.Value,
+				Siblings:   voter.proof.Proof,
+				LeafWeight: voter.proof.LeafValue,
 			},
 		},
 	}
+	stx := models.SignedTx{}
 	stx.Tx, err = proto.Marshal(&models.Tx{Payload: &models.Tx_Vote{Vote: vote}})
 	qt.Assert(t, err, qt.IsNil)
-	stx.Signature, err = voterKey.SignVocdoniTx(stx.Tx, server.VochainAPP.ChainID())
+	stx.Signature, err = voter.key.SignVocdoniTx(stx.Tx, te.server.VochainAPP.ChainID())
 	qt.Assert(t, err, qt.IsNil)
-	stxb, err = proto.Marshal(&stx)
+	stxb, err := proto.Marshal(&stx)
 	qt.Assert(t, err, qt.IsNil)
 
 	v := &api.Vote{TxPayload: stxb}
-	resp, code = c.Request("POST", v, "votes")
+	resp, code := te.c.Request("POST", v, "votes")
+	if code == 500 {
+		t.Logf("%s", resp)
+	}
 	qt.Assert(t, code, qt.Equals, 200)
 	err = json.Unmarshal(resp, &v)
 	qt.Assert(t, err, qt.IsNil)
+	return v
+}
 
-	// Block 3
-	server.VochainAPP.AdvanceTestBlock()
-	waitUntilHeight(t, c, 3)
-
-	// Verify the vote
-	_, code = c.Request("GET", nil, "votes", "verify", election.ElectionID.String(), v.VoteID.String())
-	qt.Assert(t, code, qt.Equals, 200)
-
-	// Get the vote and check the data
-	resp, code = c.Request("GET", nil, "votes", v.VoteID.String())
-	qt.Assert(t, code, qt.Equals, 200)
-	v2 := &api.Vote{}
-	err = json.Unmarshal(resp, v2)
-	qt.Assert(t, err, qt.IsNil)
-	qt.Assert(t, v2.VoteID.String(), qt.Equals, v.VoteID.String())
-	qt.Assert(t, v2.BlockHeight, qt.Equals, uint32(2))
-	qt.Assert(t, *v2.TransactionIndex, qt.Equals, int32(0))
+func (te *testElection) VerifyVotes(t testing.TB) {
+	for _, voter := range te.voters {
+		// Verify the vote
+		_, code := te.c.Request("GET", nil, "votes", "verify",
+			te.election.ElectionID.String(), voter.vote.VoteID.String())
+		qt.Assert(t, code, qt.Equals, 200)
 
-	// TODO (painan): check why the voterID is not present on the reply
-	//qt.Assert(t, v2.VoterID.String(), qt.Equals, voterKey.AddressString())
+		// Get the vote and check the data
+		resp, code := te.c.Request("GET", nil, "votes", voter.vote.VoteID.String())
+		qt.Assert(t, code, qt.Equals, 200)
+		v2 := &api.Vote{}
+		err := json.Unmarshal(resp, v2)
+		qt.Assert(t, err, qt.IsNil)
+		qt.Assert(t, v2.VoteID.String(), qt.Equals, voter.vote.VoteID.String())
+		qt.Assert(t, v2.BlockHeight, qt.Equals, uint32(2))
+		qt.Assert(t, v2.VoterID.String(), qt.Equals, fmt.Sprintf("%x", voter.key.Address().Bytes()))
+	}
 }
 
 func TestAPIaccount(t *testing.T) {
```

Comment:
there's a lot of code in TestAPIaccount that could be deduplicated now, after the proposed refactoring. i'd include this in the same PR